### PR TITLE
- document CMAKE_GENERATOR_PLATFORM

### PR DESCRIPTION
**Pull request by SSE4**
for https://github.com/conan-io/conan/issues/4707 https://github.com/conan-io/conan/issues/4755 https://github.com/conan-io/conan/pull/4708

### CODE REVIEW

File: reference/build_helpers/cmake.rst [hunk 1]
```diff
@@ -65,10 +66,23 @@ Parameters:
...
+    - **generator_platform** (Optional, Defaulted to ``None``): Generator platform name or none to autodetect (-A cmake option).
 
 Attributes
 ----------
 
+generator
++++++++++
+
+Specify a custom CMake generator to use, see also `cmake-generators documentation <https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html>`.
+If used before the ``configure()`` call, overrides the generator to use. If used after the ``configure()`` call, reports the actual generator in use (e.g. auto-detected one).
```
Comment (danimtb) [inline reference/build_helpers/cmake.rst]: Not sure what you mean by "reports the actual generator". Is there a difference if you read this variable before and after the configure() call?

Comment (SSE4) [reply reference/build_helpers/cmake.rst]: oh, you're right, auto-detection is done in ctor, not in configure, I'll change it

**Comment by SSE4** [comment]
@danimtb this one seems to be get missed

**Comment by danimtb** [comment]
Totally right @SSE4... Sorry about that. Could you please cherry-pick the commit and make the change in the master branch? I would like to have it included in the current 1.14 documentation

**Comment by danimtb** [comment]
Thanks @SSE4

**Pull request merged**